timely_beliefs/beliefs/classes.py

```diff
@@ -182,17 +183,15 @@ class TimedBeliefDBMixin(TimedBelief):
 
     @declared_attr
     def __table_args__(cls):
-        if has_inherited_table(cls):
-            return (
-                UniqueConstraint(
-                    "event_start",
-                    "belief_horizon",
-                    "sensor_id",
-                    "source_id",
-                    name="_one_belief_by_one_source_uc",
-                ),
-            )
-        return None
+        return (
+            Index(
+                f"{cls.__tablename__}_quad_search_idx",
+                "event_start",
+                "source_id",
+                "sensor_id",
+                "belief_horizon",
```

Comment:
Thanks. I read somewhere that range columns (like event_start) are a good place to lead with, as well. 

To be honest, I will simply add the query that Mike suggested in the end, but not because I see a performance difference. After #166, the performance for sensors in our dataset with little data is so fast, I suspect indexes are not visible. And for the sensor with > 50% of the data, Postgres seems to ignore the index by default.

Not listing belief_horizon in the index (but including it as column) makes sense, as we are using `min()` on it.

This PR is an improvement over the status quo, but we might revisit indexing when we have much more data.